client/app/bundles/KudosApp/reducers/appReducer.jsx

```diff
@@ -145,13 +146,21 @@ const initialize = (
   return state
 }
 
+const activateModal = (state = false, action) => {
+  if (action.type === actionTypes.MODAL_SWITCH) {
+    return !state
+  }
+  return state
+}
```

Comment:
Does `activate` mean the modal is shown? (See my naming comment above.)

Also, for returning results of conditionals, you can simplify by return the statement itself! (Though you'll have to reverse it here.)
```js
return action.type !== actionTypes.MODAL_SWITCH
```